### PR TITLE
fix(learn): in ES6 spread operator challenge test for arr2 length > 0

### LINKED ISSUE #39662: ES6: Use the Spread Operator to Evaluate Arrays In-Place
**Describe your problem and how to reproduce it:**
In the challenge that I mentioned in the title there is an issue. Even when the output doesn't match with the challenge requirements but it still passes the challenge. I have linked some gifs so it is more understandable.

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/es6/use-the-spread-operator-to-evaluate-arrays-in-place

**Tell us about your browser and operating system:**
* Browser Name: Google Chrome
* Browser Version: 85.0.4183.121
* Operating System: Windows 10 64-bit

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**
The output doesn't match with the requirements but it still passes the challenge:
https://gyazo.com/f27f86327cd9dc9d5959c2dc456f1041

While this is the correct code and output:
https://gyazo.com/0d11507fed0226112f059f45c0e5b97f

This also passes the challenge:
https://gyazo.com/2e3ae5310e4b638b0156ffc174b2e155

### PR DESCRIPTION
**Pull request by walterpalagi**
Checklist:

- [ x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [ x] My pull request targets the `master` branch of freeCodeCamp.
- [ x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #39662

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-the-spread-operator-to-evaluate-arrays-in-place.english.md
Comment (nhcarrigan) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-the-spread-operator-to-evaluate-arrays-in-place.english.md:46]:
```suggestion
    testString: assert(arr2.every((v, i) => v === arr1[i]) && arr2.length);
```

Because `0` is falsy, we can assert for the truthiness of the `arr2.length` value. 🙂

**Review by naomi-lgbt — COMMENTED** [review]
Congratulations on your first PR and thank you for your interest in contributing! 🥳 

I have just one slight nitpick.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-the-spread-operator-to-evaluate-arrays-in-place.english.md
Comment (walterpalagi) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-the-spread-operator-to-evaluate-arrays-in-place.english.md:46]: That's right. Since I have to go back to my code months after writing it when a customer asks me to expand on it, I always go for the most explicit implementation unless there's a difference in performance - it makes picking it up again easier. It's gotten to the point where I do it without even thinking about it. Your correction is (of course) more in line with the style used in this project. Thanks!

**Review by naomi-lgbt — APPROVED** [review]
LGTM! Our workflow requires at least one additional approval before this can be merged. 🙂

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
Congrats @walterpalagi on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Pull request merged**
